### DISCUSSION TITLE
Web Follower for Libre 3 returning 403 error

### DISCUSSION THREAD

**Discussion by ljlongwing** [NightscoutFoundation/xDrip#2893] [Q&A] [answered]
Earlier this week Libre did some work on their servers and when they came back up my X-Drip stopped working.  Doing some troubleshooting today I looked at the system status and the message returned is
Got exception 2: java.lang.RuntimeException: 403 Got failure response code: 403 {"data": {'minimumVersion":"4.7.0"},"status":920}
I found I was running an old version so I upgraded to the May 'latest' release, the problem did not resolve.  I then upgraded to the latest nightly build, which as of this moment is 745d695-2023.06.15.  This also did not resolve anything.  I went into my Libre 3 App and Connected Apps and removed the follower that xDrip uses, then re-added them, then logged back into the LibreLinkUp and re-accepted the invite.  Everything from the Libre side is working fine, all numbers are displaying as expected, it's only the xDrip app that's having trouble getting in.
![Screenshot_20230617_063453_xDrip+](https://github.com/NightscoutFoundation/xDrip/assets/9967628/5b16f800-fde4-4da3-a70f-3ef8dbb4b02d)

**Comment by Navid200** [comment]
Did you see this?
https://github.com/NightscoutFoundation/xDrip/discussions/2888

**Reply by ljlongwing** [reply]
it might, if I understood where I needed to update that URL....I don't see where in that discussion is says that URL is in the interface

**Reply by ljlongwing** [reply]
Ok...so, yes, sort of.....Current process is to use Web Follower specifying the ID/PW you use in LibreLinkUp....but, this new process is to install another, 3rd party tool found https://github.com/cmtjk/LinkUpConnect, then in that app, provide the same ID/PW as you would have for WebFollower, but change the server URL from api-de.libreview.io to something specific to your region, in my case api-us.libreview.io, then check the box that says 'Froward to XDrip, and go BACK into xDrip and change my 'Hardware Data Source' from 'Web Follower to 'Libre (patched app)'.....this seems to have worked for now.....but doesn't seem to be a 'fix' for the original configuration, instead, a workaround.  Thank you for pointing me to that discussion, I did search, but didn't find it based on the keywords I was using.

**Reply by jamorham** [reply]
in xDrip try adding a space to the end of the `Configuration Script` web follower setting and see if this helps at all.

**Reply by ljlongwing** [reply]
I added a space after 'beonlabs' to be 'beonlabs ' and that still reports the same error as initially

**Reply by hpcolus** [reply]
webfollow for freestyle libre is not possible anymore after abbot's update to 4.7 version.  The interim solution, not ideal, is to change in xDrip the hardware set up/source  as libre patched, and install a linkup connect apk.  It is in this application, linkup connect, where you need to enter the location (url) specific to your region, not in xDrip.

**Reply by jamorham** [reply, marked as answer]
@ljlongwing if you remove the space again do you still get the same status message?

**Reply by dxnxz** [reply]
@jamorham that did the trick!!! seems to work again. the "latest error" is about 10 minutes ago, right after i switched back to webfollower as a source. adding and then removing the space after beonlabs seems to have helped! since then i had 2 successful readings

**Reply by ljlongwing** [reply]
Yes, I can now confirm, on two separate phones, one of which I had to log out and back in to accept the license agreement, but on both of them, putting a space, waiting, removing the space seems to have fixed it, don't understand that, but it seems to have worked

**Reply by jamorham** [reply]
@ljlongwing doing those steps causes a full reset of the internal state.

**Reply by jeveli** [reply]
Yes this works.

**Reply by kexana** [reply]
This is still a working fix!

**Comment by danielrealm24** [comment]
Where is Configuration Script in xdrip app? Thank you

**Reply by tqkam** [reply]
It was loaded from https://community-script.beonlabs.net, but it is not available now.

**Comment by mcassiopeia** [comment]
Hello,
I got the same problem since this morning only with version 4.12.0. I already tried the space solution a couple of times but it does not work. I also deinstalled the librelinkup app and installed it new in order to get the "new" version. The connection between Libre and the Linkup works fine only Xdrip has a problem. I already tried to install Xdrip on my tablet but got the same error.

I already found a page "Stelo improvements" where a source code is added that promises to solve this problem but I sadly have no Idea how to install it (I normally install the apk and I did not find any in the downloaded zip).

Does anybody have the same problem at the moment? Maybe Abbot has changed something on the Linkup System or do you have another idea how I could solve this problem?

**Reply by NikTiddy** [reply]
Since Yesterday I have the same problem while using my G-Watch App:
Sending LibreLinkUp session request. FAILED: HTTP 403

Same error occurs in xdrip too unfortunately

**Reply by pachi81** [reply]
> Since Yesterday I have the same problem while using my G-Watch App: Sending LibreLinkUp session request. FAILED: HTTP 403
> 
> Same error occurs in xdrip too unfortunately

There is already a solution for this issue.
In the meantime you can use GlucoDataHandler, which can forward the values to xDrip+ and also G-Watch:
https://github.com/pachi81/GlucoDataHandler/releases/tag/1.2.1

**Comment by pachi81** [comment]
If you have not already found a solution,
you need to set version 4.12.0 in header and add the user id as sha256 hex to header:
https://github.com/JohanDegraeve/xdripswift/commit/13e4a1e7418364fbdb68e5bd54cac888fd2eb9b8

**Reply by CHSTPH** [reply]
Is there an apk available that contains this fix?

**Reply by HoLi1969** [reply]
Hello, is there a description how to change the version and header? In the referenced post it is shown but I have no idea where to find the 3 files and how to edit.

**Reply by HoLi1969** [reply]
Nightly from yesterday does not solve the issue

**Reply by florianAIO** [reply]
Good to know. I was also irritated by the "There is already a solution for this issue."

**Comment by PhilTU** [comment]
@Navid200 did you see this. Added to "answered" topic but new problem has arisen as Abbott have changed minimum version of Librelinkup hence script or something needs updating to comply.

**Reply by Navid200** [reply]
This is not something I can do anything about other than informing the developers.  I will do that now.  Thanks

**Reply by HoLi1969** [reply]
Hi, my xDrip Web Flower also stopped working yesterday. I attach the system status error message. Is this the same issue? 
![Screenshot_2024-11-16-22-54-54-90_2dd2c1bcb8f659cd696bcba2072f751b](https://github.com/user-attachments/assets/32b4f4c0-2011-45dd-85e4-af615d3a9fa6)

**Reply by pachi81** [reply]
> Hi, my xDrip Web Flower also stopped working yesterday. I attach the system status error message. Is this the same issue? ![Screenshot_2024-11-16-22-54-54-90_2dd2c1bcb8f659cd696bcba2072f751b](https://private-user-images.githubusercontent.com/110985645/386880873-32b4f4c0-2011-45dd-85e4-af615d3a9fa6.jpg?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3MzE3OTUzMTksIm5iZiI6MTczMTc5NTAxOSwicGF0aCI6Ii8xMTA5ODU2NDUvMzg2ODgwODczLTMyYjRmNGMwLTIwMTEtNDVkZC04NWU0LWFmNjE1ZDNhOWZhNi5qcGc_WC1BbXotQWxnb3JpdGhtPUFXUzQtSE1BQy1TSEEyNTYmWC1BbXotQ3JlZGVudGlhbD1BS0lBVkNPRFlMU0E1M1BRSzRaQSUyRjIwMjQxMTE2JTJGdXMtZWFzdC0xJTJGczMlMkZhd3M0X3JlcXVlc3QmWC1BbXotRGF0ZT0yMDI0MTExNlQyMjEwMTlaJlgtQW16LUV4cGlyZXM9MzAwJlgtQW16LVNpZ25hdHVyZT1kMTA4ZGQ3NzU4ZGJmZDA0ZjQ2NjBhOWYzYjgyZmU4ODkyNzZjMmRjMmRmNWRjY2NhMjFlYWQ5OWE0MWEyOWY2JlgtQW16LVNpZ25lZEhlYWRlcnM9aG9zdCJ9.gGIiZIgSIY6_WZ7V8fD-9M8ZYS31wggAi9jB7n42sDA)

No, you have to accept new terms of user in the LibreLinkUp app. Just re-install LibreLinkUp app, login and accept terms.

**Reply by HoLi1969** [reply]
Good morning, thanks. I re-installed librelinkupp app and accepted conditions. Now I get a 403 error. I tried already different changes I found here regarding 403 but still no change.
![Screenshot_2024-11-17-08-22-39-12_2dd2c1bcb8f659cd696bcba2072f751b](https://github.com/user-attachments/assets/440cbeff-0eda-4a82-afec-93b0ab023aa6)

**Reply by pachi81** [reply]
https://github.com/NightscoutFoundation/xDrip/discussions/2893#discussioncomment-11276518

**Reply by HoLi1969** [reply]
Thanks, unfortunately I am not a programmer, just stupid user. I understand there are entries to be done under developer settings in xDrip? Unfortunately I cannot activate such settings. Would be great to get some instructions "Barney style".

**Reply by mcassiopeia** [reply]
Hello,
I got the same problem but the solution with the GlucoDatHandler is very easy. Just install the Program "GlucoDatHandler". You can download it over Google Play Store. Then add you LibrelinkUp Data to the program and it will receive your values frol Librelikup. Over settings you can send them to X-drip (dierectly over the phone). You have to change tha Data-Source in X-Drip to libre (patched) and it will receive your data.

Just take care to allow both apps (xdrip and GlucoDatHandler) to work in the background or it will stop to send data after some time.

**Reply by HoLi1969** [reply]
Unfortunately I do not get it to work. GlucoDatHandler receives a first value and then no further.

**Reply by pachi81** [reply]
> Unfortunately I do not get it to work. GlucoDatHandler receives a first value and then no further.

Is there any error? Have you disabled battery optimization and so on? Enabled notification? You can use the contact option  to contact me direct.

**Reply by HoLi1969** [reply]
Unfortunately the last Nightly did not solve the issue
![Uploading Screenshot_2024-11-18-11-31-08-89_2dd2c1bcb8f659cd696bcba2072f751b.jpg…]()

**Comment by mcassiopeia** [comment]
The GlucoDataHandler solution worked for me. Thanks a lot for your answers and for the help!

**Comment by frietbakje** [comment]
Glucodatahandler werkt idd, kreeg in xdrip ook enige tijd die 403 foutmelding die niet op te lossen is. Weet niet of dit weer een Belgisch probleem is of als er een oplossing is om enkel xdrip te gebruiken.

**Reply by frietbakje** [reply]
Glucodatahandler is working fine, but I also got a 403 error in xdrip that I can't resolve. I'm not sure if this is another Belgian issue or if there's a solution to use only xdrip.

**Reply by frietbakje** [reply]
You can update xdrip to the latest version, put a space after Beonlabs and then delete it again, install the LibreLinkUp app ... It still doesn't work.

**Comment by HoLi1969** [comment]
Hi, no, I s. On it since last year and my understanding is that the new version of Librelinkup must be put into the code. I found a description for the same issue in the past but it seems that no programmer will do it anymore. I have no idea how to work on the code and to rebuild the APK.

Gesendet von Outlook für Android<https://aka.ms/AAb9ysg>
________________________________

**Reply by frietbakje** [reply]
Thanks, I assume you are referring to: https://github.com/JohanDegraeve/xdripswift/release

**Comment by HoLi1969** [comment]
As I said, the version needs to be updated to 4.12

Gesendet von Outlook für Android<https://aka.ms/AAb9ysg>
________________________________

**Reply by frietbakje** [reply]
Thanks, I assume you are referring to: https://github.com/JohanDegraeve/xdripswift/release
I used to be able to convert code into Java, but it wasn't the programming language I worked with...
See what I can do.

**Reply by frietbakje** [reply]
I have checked the following pages because the previous ones apparently don't work, 
but it seems that it no longer works and it's best to use GlucoHandler. 
https://github.com/JohanDegraeve/xdripswift/releases?page=1 
https://github.com/NightscoutFoundation/xDrip/discussions/3808

**Comment by jamorham** [comment]
If someone wants to share their link up username and password with me I can take a look. Please send to my email jamorham at gmail

**Comment by jamorham** [comment]
A fix for this should be coming soon but it would be helpful if more people could share credentials with me via email so I can test it. So far I have tested for germany.

**Reply by PhilTU** [reply]
Do you have enough?

**Reply by jamorham** [reply]
No, I really need some people to share follower logins for me, specifically for regions outside of germany. If anyone can help please contact me via email. jamorham at gmail

**Reply by Navid200** [reply]
I have announced this on facebook.  If they don't contact you, they cannot complain any more.  I am going to announce again now.

**Reply by PhilTU** [reply]
> No, I really need some people to share follower logins for me, specifically for regions outside of germany. If anyone can help please contact me via email. jamorham at gmail

I will send you one in a few minutes, you already have my German one. There may be no active data but I guess this is not important as it's an authentication issue.

**Comment by HoLi1969** [comment]
Great to hear.

Gesendet von Outlook für Android<https://aka.ms/AAb9ysg>
________________________________

**Comment by jamorham** [comment]
Please try the latest nightly version to see if it resolves the issue. It works with the two regions I have tested with so far. Special thanks to those users who shared credentials with me to test with.

**Reply by Epsolone** [reply]
I still have the same issue in Germany. App was updated. LibreLinkUp account works and received data. How I can resolve it?

**Reply by Navid200** [reply]
@Epsolone Post a screenshot of the classic status page.

**Reply by Epsolone** [reply]
@Navid200 
![photo_5305508620861042563_y](https://github.com/user-attachments/assets/497d38a4-f305-497c-802f-59d95204587f)

**Reply by Navid200** [reply]
You can see that you are running the 2024.09.17 release.  You can see the date right after Version at the top.  

Go to release notes here: https://navid200.github.io/xDrip/docs/ReleaseNotes.html 

You can see that we updated Web follower on 2025.02.07.  Click on that link in the left column.  It will take you to the release.  

Download and install the file with .apk at the end to update.

**Reply by Epsolone** [reply]
@Navid200 Thanks. I updated with 2025.02.07 release version and now all works fine.

**Comment by HoLi1969** [comment]
Hi,
I can confirm that xDrip Web Follower works again. So for Germany it looks fine.

Many thanks. xDrip really is our backbone and the best available. I am really thankful for you and the other experts for your effort and support.

Have a nice weekend!

Gesendet von Outlook für Android<https://aka.ms/AAb9ysg>
________________________________

**Comment by HoLi1969** [comment]
Good morning,

I need to ask again for help. Seems to be the same issue with an update of Librelinkup App. New version is no longer supported after 8th so last night xDrip no longer gets data through Web Follower. I attach a screenshot of the new Librelinkup version.

Are you able to make the necessary changes in nightly update of xDrip? Our whole alert chain depends on xDrip.
Or is there a way I can solve the issue on my end?

Many thanks,
Holger Skov

Gesendet von Outlook für Android<https://aka.ms/AAb9ysg>
________________________________

**Reply by PhilTU** [reply]
Also reported at https://github.com/NightscoutFoundation/xDrip/discussions/4203

@Navid200 another Abbott update requires minimum version 4.16 so a script change required once again in the xDrip ecosystem 

This is a long term vulnerability I wouldn't base anything on the LinkUp interface with "Libre by Abbott" app coming down the road.
